example_test.go

```diff
@@ -62,6 +64,54 @@ func Example() {
 	}
 }
 
+// This example to test event broadcaster to set qps and burst size.
+func ExampleEventBroadcaster() {
+	var examplePod = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example",
+			Namespace: "default",
+		},
+	}
+	var log = ctrl.Log.WithName("event-examples")
+	// Controller have two diff recorderProvider
+	// 1.manager will create recorderProvider to resourceLock and push lease event.
+	// 2.cluster also create recorderProvider,when call manager.GetEventRecorderFor,will get the provider.
+	// When use the provider to push event like eventf(),provider will call getBroadcaster and StartEventWatcher.
+	// If use the EventBroadcaster options,the diff recorderProvider's makeBroadcaster func will return same broadcaster point
+	// Then call broadcaster.StartEventWatcher func duplicate,event will dunplicate push.
+	manager, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		LeaderElectionID:        "123456qbcdqe",
+		LeaderElection:          true,
+		LeaderElectionNamespace: "test1",
+		EventBroadcaster: record.NewBroadcasterWithCorrelatorOptions(record.CorrelatorOptions{
+			BurstSize: 50,
+			QPS:       15,
+		}),
+	})
+	if err != nil {
+		log.Error(err, "could not create manager")
+		os.Exit(1)
+	}
+	go func() {
+		for {
+			select {
+			case <-ctrl.SetupSignalHandler().Done():
+				return
+			case <-time.After(time.Second * 10):
+				er := manager.GetEventRecorderFor("event-test")
+				// Like this,will push the same event two times.
+				er.Eventf(examplePod, corev1.EventTypeNormal, "event-test", "EventBroadcaster test")
+				log.V(1).Info("eventf end")
+			}
+		}
+	}()
+	if err := manager.Start(ctrl.SetupSignalHandler()); err != nil {
+		log.Error(err, "could not start manager")
+		os.Exit(1)
+	}
+	log.Info("success")
+}
+
```

Comment:
if not fix and use the example.
will see this log.
one times 'eventf end' log and two times 'EventBroadcaster test' log.
```
I0221 11:40:46.893743   93446 leaderelection.go:258] successfully acquired lease test1/123456qbcdqe
2023-02-21T11:40:46+08:00	DEBUG	events	recorder/recorder.go:105	niuyueyangdeMacBook-Pro.local_28a6f8b8-3c7a-478c-afde-7d46495999fc became leader	{"type": "Normal", "object": {"kind":"Lease","namespace":"test1","name":"123456qbcdqe","uid":"5f7083cc-315a-4888-bacd-66e9b0e6fe78","apiVersion":"coordination.k8s.io/v1","resourceVersion":"14055689"}, "reason": "LeaderElection"}



2023-02-21T11:40:56+08:00	DEBUG	event-examples	main/main.go:58	eventf end
2023-02-21T11:40:56+08:00	DEBUG	events	recorder/recorder.go:105	EventBroadcaster test	{"type": "Normal", "object": {"kind":"Pod","namespace":"default","name":"example","apiVersion":"v1"}, "reason": "event-test"}
2023-02-21T11:40:56+08:00	DEBUG	events	recorder/recorder.go:105	EventBroadcaster test	{"type": "Normal", "object": {"kind":"Pod","namespace":"default","name":"example","apiVersion":"v1"}, "reason": "event-test"}
```

Reply:
Thanks for the example, I'm wondering though if we can add a failing test to the manager that makes sure events don't get duplicated